internal/daysteps/daysteps.go

```diff
@@ -1,19 +1,71 @@
 package daysteps
 
 import (
-	...
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/Yandex-Practicum/go1fl-sprint5-final/internal/personaldata"
 )
 
+// StepLength — средняя длина шага в метрах.
 const (
 	StepLength = 0.65
 )
 
-// создайте структуру DaySteps
-...
+var (
+	ErrWrongDataFormat = errors.New("wrong length of data string")
+	ErrConvSteps       = errors.New("can't convert steps to int")
+	ErrParseDuration   = errors.New("can't parse duration")
+)
+
+// DaySteps содержит данные о дневных прогулках.
+type DaySteps struct {
+	Steps    int
+	Duration time.Duration
+	personaldata.Personal
+}
+
+// Parse парсит строку формата "steps,duration" (например, "678,0h50m")
+// и сохраняет результат в структуру DaySteps.
+func (ds *DaySteps) Parse(datastring string) (err error) {
+	parts := strings.Split(datastring, ",")
+	if len(parts) != 2 {
+		return ErrWrongDataFormat
+	}
+
+	// Парсим количество шагов
+	steps, err := strconv.Atoi(strings.TrimSpace(parts[0]))
+	if err != nil {
+		return ErrConvSteps
```

Comment:
Теряешь сообщение об ошибки
```suggestion
		return fmt.Errorf("conversion error: %w", err)
```

Reply:
Resolved
Changed to ```return fmt.Errorf("%w: %v", ErrConvSteps, err)```